src/RemoteRequest/CachedRemoteGetRequest.php

```diff
@@ -135,7 +135,11 @@ public function get( $url ) {
 
 			$cached_response = new CachedResponse( $body, $headers, $status, $expiry );
 
-			set_transient( $cache_key, serialize( $cached_response ) ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
+			// Transient extend beyond cache expiry to allow for serving stale content.
+			// @TODO: We don't serve stale content atm, but rather synchronously refresh.
+			// See https://github.com/ampproject/amp-wp/issues/5477.
+			$transient_expiry = $expiry->modify( "+ {$this->expiry} seconds" );
```

Comment:
Based on PHPStan ("Call to an undefined method DateTimeInterface::modify()"), I think the `get_expiry_time` return type needs to be `DateTimeImmutable` instead of `DateTimeInterface`.